libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/MetadataFuzzer.java

```diff
@@ -140,45 +148,142 @@ private void initializeFuzzer(String serviceName, ModelObject objectToBeFuzzed,
+    }
+
     this.filters = filters;
-    propertiesFuzzSet = new Object[properties.keySet().size()][3];
 
     // Prep TestNG dataprovider to make reporting better.
-    populateFuzzSet(properties, propertiesFuzzSet, "rootProperty");
+    root = (objectToBeFuzzed.getModelType() == ModelPropertyType.OBJECT) ? File.separator
+        : File.separator + "0" + File.separator ;
+
+    propertiesFuzzSet = populateFuzzSet(this.entity, new ArrayList<Object[]>(), root, this.entity.getModel());
+
+    for (int i = 0; i < propertiesFuzzSet.length; i++) {
+      System.out.println(propertiesFuzzSet[i][0] + ", " + propertiesFuzzSet[i][1] + ", "
+          + propertiesFuzzSet[i][2]);
+    }
+    
+    ModelPropertyType type = entity.getModelType();
+    if (type == ModelPropertyType.ARRAY) {
+      ModelObject.prettyPrintToConsole(entity.getObjectItems());
+    } else {
+      ModelObject.prettyPrintToConsole(entity.getObjectMetadata());
+    }
+  }
+
+
+  private String getFuzzPath(JSONObject currentPathModel, String currentPath, String propertyName) {
+    String fuzzPath = "";
+    try {
+      ModelPropertyType type = ModelObject.getModelPropertyType(currentPathModel);
+      if (type == ModelPropertyType.ARRAY) {
+        fuzzPath = currentPath + propertyName + File.separator + "0" + File.separator;
+      } else {
+        fuzzPath = currentPath + propertyName + File.separator;
+      }
+    } catch (UnexpectedModelPropertyTypeException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+
+    return fuzzPath;
+  }
+  
+  private class ModelProperties{
+    Object[] propertiesArray;
+    JSONObject properties;
+    
+    public ModelProperties(JSONObject properties) {
+      this.properties = properties;
+      this.propertiesArray = properties.keySet().toArray();
+    }
+  }
+
+  private ModelProperties getObjectProperties(ModelObject currentfuzzObject,
+      JSONObject currentPathModel) {
+    JSONObject properties = null;
+
+    properties = ModelObject.getObjectNodeProperties(currentfuzzObject, currentPathModel);
+
+    return new ModelProperties(properties);
+  }
+
+  private boolean isObjectProperty(ModelPropertyType type) {
+    if (type == ModelPropertyType.OBJECT || type == ModelPropertyType.ARRAY
+        || type == ModelPropertyType.REF_SCHEMA || type == ModelPropertyType.REF_DEFINITION) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
+
   /**
    * Method which turns the properties of the model into a dataprovider for the fuzzer to use.
    * 
    * @param set JSONObject representing a set of metadata nodes
    * @param fuzzSet the array in which the data will be pushed
-   * @param propertyType indicator specifying links or root property
+   * @param fuzzPropertyType indicator specifying links or root property
    * @throws UnexpectedModelPropertyTypeException
    */
-  private void populateFuzzSet(JSONObject set, Object[][] fuzzSet, String propertyType) {
```

Comment:
Changes in this method are all related to recursion and getting all possible fuzzable paths.